Module2/task2/server.js

```diff
@@ -0,0 +1,114 @@
+const express=require('express')
+const fs=require('fs');
+const { Http2ServerRequest } = require('http2');
+const path=require('path')
+const app=express();
+const bodyParser=require('body-parser')
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.get('/users',(req,res)=>{
+fs.readFile(path.join(__dirname,'user.json'),'utf8',(err, data)=>{
+    if(err){
+        console.log(err);
+    }else
+    res.json(JSON.parse(data));
+});
+});
+
+ app.get('/users/:id/user',(req,res)=>{
+    console.log(req.params.id);
+    fs.readFile(path.join(__dirname,'user.json'),'utf8',(err, data)=>{
+        if(err){
+            console.log(err);
+        }else
+        {
+            const output=JSON.parse(data);
+            for(let obj of output.users){
+                if(obj.id==req.params.id){
+                    res.send(obj);
+                }
+            }
+        }     
+    });
+});
+
+app.delete('/users/:id', (req, res) => {
+    const userId = req.params.id;
+    // console.log(userId);
+    fs.readFile(path.join(__dirname, 'user.json'), 'utf8', (err, data) => {
+    const input=JSON.parse(data);
+    for (let i in input.users){
```

Comment:
Why we are not using Higher order function here?

Reply:
Updated